src/e2e/puppeteer/helpers/setSelection.ts

```diff
@@ -0,0 +1,22 @@
+import { page } from '../setup'
+
+/** Sets the selection of the text in the editable. */
+const setSelection = async (start: number, end: number) => {
+  await page.evaluate(
+    (start: number, end: number) => {
+      const selection = window.getSelection()
+      const range = document.createRange()
+      const textNode = document.querySelector('[contenteditable="true"]')?.firstChild
```

Comment:
This will only set the selection on the cursor thought if it is the first thought.

To make this more flexible, use `document.querySelector('[data-editing=true] [data-editable]')`.